src/Shared/FileMatcher.cs

```diff
@@ -780,7 +780,7 @@ private static void GetFilesRecursive
                         for (int i = 0; i < excludeNextSteps.Length; i++)
                         {
                             if (excludeNextSteps[i].Subdirs != null &&
-                                excludeNextSteps[i].Subdirs.Any(excludedDir => FileUtilities.PathsEqual(excludedDir, subdir)))
+                                excludeNextSteps[i].Subdirs.Any(excludedDir => excludedDir == subdir))
```

Comment:
Do we need to avoid linq here? How hot is this path?

Reply:
Pretty hot. I'll go ahead and remove all Linq operations from this method

Reply:
On the other hand, @maca88's changes remove the need for this linq operation, so best to leave like this to reduce the future merge conflicts :)